### ISSUE TITLE
Migrate Observatory service tests to package:vm_service test suite

### ISSUE THREAD

**Issue by bkonyi** [dart-lang/sdk#45037]
Test coverage for `package:vm_service` is quite sparse. The Observatory service tests should be migrated to use `package:vm_service` to improve test coverage and prepare for the eventual Observatory deprecation.

**Comment by a-siva** [comment]
Is this migration done?

**Comment by a-siva** [comment]
If there are a few tests still pending which are not as critical maybe we can drop the P2 label

**Comment by bkonyi** [comment]
Yeah, there's still a couple of tests left that are more tricky to migrate (e.g., tests that involve building profile trees to verify the contents of a `CpuSamples` response). Maybe we should consider just not migrating them, but in the meantime we can drop the priority to P3.
